src/components/ThemeEngine/ThemeEngine.constants.ts

```diff
@@ -0,0 +1,144 @@
+export enum Theme {
+  Light = 'light',
+  Dark = 'dark'
+}
+
+export const DEFAULT_THEME = Theme.Dark;
+
+export enum ThemeColorKeys {
```

Comment:
Can we change this enum to `zUITheme`?

Reply:
Hmm yeah it's a tough one.

We may actually not even need the `ThemeColorKeys` enum - it might be cleaner to do:
```js
type Theme = {
    primary1: string;
    ...
}

const Dark: Theme = {
    primary1: "#000"
    ....
}

// converting to CSS vars (ignore the values, just communicating the idea)
const keys = Object.keys(Dark);
keys.forEach(key => {
    const css = `zui-${key}-${Dark[key]}`
}
```
The above might make things a little more clear, I think having the separate enum just for the color keys might be redundant.

Reply:
Agree! But we have `Theme` enum which is `dark` or `light`.  I think we need to use another type `Theme`.